### PR TITLE
Fix chart rendering edge cases and hover styles

### PR DESCRIPTION
**Pull request by bru5**
## Description

Fixes two rare errors with chart rendering, that are likely due to timing edge cases.

Also fixes sidebar hover styling for disabled buttons.

## Development notes

As above.

## QA notes

The errors themselves are hard to reproduce, but they seem to mostly occur in Safari, after rapidly applying filters on the sidebar in random order.

To test the sidebar issue, load a graph with no layers such that the layers button is disabled. Hovering should no longer cause this or other disabled buttons to highlight.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: src/components/app/app.scss [hunk 1]
```diff
@@ -13,10 +13,6 @@
   .kui-theme--dark & {
     fill: $color-light;
   }
-
-  .pipeline-icon-toolbar__button:hover & {
-    opacity: 1;
-  }
```
Comment (richardwestenra) [inline src/components/app/app.scss:19]: Why this change? Will this not totally remove all hover styling? If you want to avoid having hover states on disabled buttons, I think this would be a better approach:
```scss
.pipeline-icon-toolbar__button:not(:disabled):hover & {
  opacity: 1;
}
```

File: src/components/flowchart/draw.js [hunk 2]
```diff
@@ -180,8 +180,12 @@ export const drawNodes = function(changed) {
       .duration(this.DURATION)
       .attr('transform', node => `translate(${node.x}, ${node.y})`)
       .on('end', () => {
-        // Sort nodes so tab focus order follows X/Y position
-        allNodes.sort((a, b) => a.order - b.order);
+        try {
+          // Sort nodes so tab focus order follows X/Y position
+          allNodes.sort((a, b) => a.order - b.order);
+        } catch (ex) {
```
Comment (richardwestenra) [inline src/components/flowchart/draw.js:186]: `ex`? Haven't seen that before, would've thought `err` usually, but whatever it really doesn't matter

File: src/components/app/app.scss (hunk 1 above)
Comment (richardwestenra) [reply src/components/app/app.scss:19]: ah I see, it's added elsewhere. Fair enough. I'm not stoked about the styles being added in two different places, but maybe we should address that in a separate PR. Nevermind :)

**Review by richardwestenra — APPROVED** [review]
Added a couple of comments, feel free to ignore them though

File: src/components/app/app.scss (hunk 1 above)
Comment (bru5) [reply src/components/app/app.scss:19]: I originally wrote the same thing, but then realised that there was no longer a need for that styling to be inside of `app.css` anymore and moved it to the icon toolbar styles where it would normally be.

There's no duplication, I'm not sure what you mean?

File: src/components/flowchart/draw.js (hunk 2 above)
Comment (bru5) [reply src/components/flowchart/draw.js]: I just tend to write `ex` for exceptions that must be handled, `err` for anything else to be clear. Anyway, I changed it to `err` then.

File: src/components/app/app.scss (hunk 1 above)
Comment (richardwestenra) [reply src/components/app/app.scss:19]: you're right, no duplication, more that the intricately bound styles are being handled in two places. Made me think that it could be done in a more obvious manner. But it's not related to your PR so ignore for now

File: src/components/flowchart/draw.js (hunk 2 above)
Comment (richardwestenra) [reply src/components/flowchart/draw.js]: aha. ok no worries, thanks!

**Review by richardwestenra — APPROVED** [review]
Once more with feeling

File: src/components/app/app.scss (hunk 1 above)
Comment (bru5) [reply src/components/app/app.scss:19]: I agree looks like it should be elsewhere too, I imagine the move to CSS variables will likely avoid what ever reason it was put here originally when it was first implemented.
